libcudacxx/include/cuda/barrier

```diff
@@ -13,4 +13,340 @@
 
 #include "std/barrier"
 
+// Forward-declare CUtensorMap for use in cp_async_bulk_tensor_* PTX wrapping
+// functions. These functions take a pointer to CUtensorMap, so do not need to
+// know its size. This type is defined in cuda.h (driver API) as:
+//
+//     typedef struct CUtensorMap_st {  [ .. snip .. ] } CUtensorMap;
+//
+// We need to forward-declare both CUtensorMap_st (the struct) and CUtensorMap
+// (the typedef):
+struct CUtensorMap_st;
```

Comment:
Most of these changes are not strictly needed for the functionality of this PR. 
It would be better to leave them out for now.